api/src/main/java/org/apache/cloudstack/api/command/admin/diagnostics/RemoteDiagnosticsCmd.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * // Licensed to the Apache Software Foundation (ASF) under one
+ * // or more contributor license agreements.  See the NOTICE file
+ * // distributed with this work for additional information
+ * // regarding copyright ownership.  The ASF licenses this file
+ * // to you under the Apache License, Version 2.0 (the
+ * // "License"); you may not use this file except in compliance
+ * // with the License.  You may obtain a copy of the License at
+ * //
+ * //   http://www.apache.org/licenses/LICENSE-2.0
+ * //
+ * // Unless required by applicable law or agreed to in writing,
+ * // software distributed under the License is distributed on an
+ * // "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * // KIND, either express or implied.  See the License for the
+ * // specific language governing permissions and limitations
+ * // under the License.
+ */
+
+package org.apache.cloudstack.api.command.admin.diagnostics;
+
+import com.cloud.exception.ConcurrentOperationException;
+import com.cloud.exception.InsufficientCapacityException;
+import com.cloud.exception.NetworkRuleConflictException;
+import com.cloud.exception.ResourceAllocationException;
+import com.cloud.exception.ResourceUnavailableException;
+import com.cloud.user.Account;
+import com.google.common.base.Preconditions;
+import org.apache.cloudstack.acl.RoleType;
+import org.apache.cloudstack.api.APICommand;
+import org.apache.cloudstack.api.ApiConstants;
+import org.apache.cloudstack.api.ApiErrorCode;
+import org.apache.cloudstack.api.BaseCmd;
+import org.apache.cloudstack.api.Parameter;
+import org.apache.cloudstack.api.ServerApiException;
+import org.apache.cloudstack.api.response.RemoteDiagnosticsResponse;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.diangosis.RemoteDiagnosticsService;
+
+import javax.inject.Inject;
+
+@APICommand(name = RemoteDiagnosticsCmd.APINAME,
+        since = "4.11",
+        description = "Execute network utility command (ping/arping/tracert from a remote host",
+        responseHasSensitiveInfo = false,
+        requestHasSensitiveInfo = false,
+        responseObject = RemoteDiagnosticsResponse.class,
+        authorized = RoleType.Admin)
+public class RemoteDiagnosticsCmd extends BaseCmd {
+
+    public static final String APINAME = "remoteDiganosis";
```

Comment:
remoteDiagnosis is a better name IMNSHO, but it must coincide with the actual API name

Reply:
Changed name to executeDiagnostics following ACS API naming conventions.